src/ActionMenu.tsx

```diff
@@ -43,12 +43,22 @@ const ActionMenuBase = ({
       renderItem({
         ...itemProps,
         role: 'menuitem',
-        onKeyPress: _event => {
-          onAction?.(itemProps as ItemProps)
+        onKeyPress: event => {
+          if (itemProps.disabled) {
+            return
+          }
+
+          const actionCallback = itemProps.onAction ?? onAction
+          actionCallback?.(itemProps as ItemProps, event)
           onClose()
         },
         onClick: event => {
-          onAction?.(itemProps as ItemProps)
+          if (itemProps.disabled) {
+            return
+          }
+
+          const actionCallback = itemProps.onAction ?? onAction
+          actionCallback?.(itemProps as ItemProps, event)
           onClick?.(event)
           onClose()
         }
```

Comment:
Instead of adding secondary key/click listeners here, can we just pass in an `onAction` and have it call `itemProps.onAction` if one exists, along with `onAction` from the `ActionMenu` props?